bridge-sdk/near-rpc-client/src/error.rs

```diff
@@ -0,0 +1,11 @@
+use near_jsonrpc_client::errors::JsonRpcError;
+
+#[derive(thiserror::Error, Debug)]
+#[error("Failed to generate Ethereum proof: {0}")]
+pub struct NearRpcError(#[source] pub Box<dyn std::error::Error>);
```

Comment:
In application code or prototypes, Result<(), Box<std::err::Error>> is acceptable, but in a library, public APIs should expose proper error types.

Reply:
In this particular case I went this way because `near_jsonrpc_client::errors::JsonRpcError` is generic. So I would have to make `NearRpcError` generic and then every error that uses it further. I figured it would be complicating the code a bit too much for a small gain. 

Using trait object here, we lose some of the branching but at least we preserve `Display` functionality. It is basically equivalent of using `anyhow` which I think is used in production sometimes?

Reply:
You can specify a default type if you already know the type of generic. If you don't know the type, unfortunately you are forced to propagate the generic up.

Reply:
Examples:
use generic:
```
#[derive(thiserror::Error, Debug)]
pub enum NearRpcError<E> {
    RpcError(#[from] JsonRpcError<E>),
}
```

or specify the types:
```
#[derive(thiserror::Error, Debug)]
pub enum NearRpcError {
    #[error("error: {0}")]
    RpcBlockError(#[from] JsonRpcError<RpcBlockError>),
    #[error("error: {0}")]
    RpcProofError(#[from] JsonRpcError<RpcLightClientProofError>),
}

```

Reply:
The second approach is what I arrived to eventually c25cff3162ba7d6c3d7e40648a9e682518d9f26f